### PR TITLE
Purge NLsolve and introduce NonlinearSolve.jl

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by avik-pal** [comment]
How does the `test_convergence` function work? I was comparing the numerical solutions to the analytical solutions and both seem to match but the tests fail :(

cc @ChrisRackauckas

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by avik-pal** [comment]
testing

@ChrisRackauckas this is how the solutions look like

**Comment by ChrisRackauckas** [comment]
No I mean plot the convergence simulation. Just use the plot recipe on it.

**Comment by avik-pal** [comment]
testing

**Comment by ChrisRackauckas** [comment]
just get rid of the last dt.

**Comment by avik-pal** [comment]
It still seems to fail

**Comment by ChrisRackauckas** [comment]
I meant the smallest dt, where the bump is.

**Comment by avik-pal** [comment]
Not sure about the `orbital.jl` tests, they converge on my local machine

**Review by YingboMa — APPROVED** [review]

**Comment by avik-pal** [comment]
I don't have commit access to the repo; someone else needs to merge!

**Comment by YingboMa** [comment]
We still cannot merge it before fixing the orbit example though

**Comment by avik-pal** [comment]
This is now ready to be merged!

**Comment by avik-pal** [comment]
Bump @ChrisRackauckas /@YingboMa

File: src/algorithms.jl
Comment (ChrisRackauckas) [inline src/algorithms.jl:2]: why autodiff false?

**Comment by ChrisRackauckas** [comment]
Just the one question on autodiff

File: src/algorithms.jl
Comment (avik-pal) [reply src/algorithms.jl:2]: I kept it false to ensure closer behavior to the NLsolve default we had previously.

Comment (ChrisRackauckas) [reply src/algorithms.jl:2]: With it being a breaking change, it would be best to get it in line with what we do elsewhere and default to forward mode AD

Comment (ChrisRackauckas) [inline src/algorithms.jl:2]:
```suggestion
const DEFAULT_NLSOLVE_SHOOTING = TrustRegion(; autodiff = Val(true))
const DEFAULT_NLSOLVE_MIRK = NewtonRaphson(; autodiff = Val(true))
```

**Pull request merged**
